collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java

```diff
@@ -146,6 +147,23 @@ public void denied() {
         });
     }
 
+    private void isAutoSendRunning() {
```

Comment:
`isAutoSendRunning` is a great name for a method that returns a boolean. For this one, maybe `startAutosendObserver` or something like that. It should be clear you're starting a process that will be ongoing during the life time of this activity. A comment might also help.

Did you deliberately show a toast here? I believe this means that if autosend is ongoing when the user first reaches the activity, they'll see the toast. Or if it starts while they're in this activity, they'll see it. They'll also see it when they tap the upload button. That's not necessarily a bad thing but I want to make sure it's deliberate. If it's intentional, there should be a comment to indicate this.

Reply:
Honestly, it doesn't have to be there. I was just thinking that when the user comes unto the screen, they could get an update that background send is already sending their forms, so they don't have to do anything. The messaging would have to be changed since it is similar to the manual send button. Was thinking to do that but was just pushing the changes I already made. If it's better they try to upload then hear that it's already running similar to the previous behaviour then that's fine with me as well.

Reply:
My sense is that it's not immediately obvious to a user why they should care so I'd tend to leave it out here.